core/block.js

```diff
@@ -1125,6 +1125,10 @@ Blockly.Block.prototype.getFieldValue = function(name) {
  * @param {!string} name The name of the field to set the value of.
  */
 Blockly.Block.prototype.setFieldValue = function(newValue, name) {
+  if (typeof name === "undefined") {
```

Comment:
Not that anyone would do this (and I'm assuming a pull request like this would fail a test anyway), but using `typeof` was to guard against someone doing something silly like prefixing the `if` statement with the following line:

```var undefined = "abc";```

Should I even be thinking like this? 🙂 As for not checking if `name` was a `string`, I was thinking of allowing future changes that might permit the developer to access fields with `name` being something other than a `string`.  E.g., in the current implementation, the following would work (because of the current use of `==` instead of `===` in `getField`):

```textBlock.setFieldValue("Blockly", {toString: () => "TEXT"})```

But I see that it'd make more sense to leave it open to changes where we'd want to treat an `object` differently.  I changed the type check, moved the error message up to `getField`, and changed the message, accordingly.  Is this what you meant?  The only concern I'd have with this is that it might obscure the error message a little when a developer uses `setFieldValue` as he/she would need to be aware that `setFieldValue` winds up calling `getField`.  Although you could always use the stack trace to figure this out, some programmers might not know this.

Reply:
> Not that anyone would do this (and I'm assuming a pull request like this would fail a test anyway), but using `typeof` was to guard against someone doing something silly like prefixing the `if` statement with the following line:

```JS
var undefined = "abc";
```

I guess I might worry about this in some particularly security-sensitive contexts, but [global `undefined` has been immutable since ES5.1](https://stackoverflow.com/a/8783528/4969945) and so it would take some considerable effort to make `undefined` be some other value.